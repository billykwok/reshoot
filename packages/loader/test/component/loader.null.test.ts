import { describe, beforeAll, afterAll, test, expect } from '@jest/globals';
import path from 'path';
import compile from './compiler';
import createMemfs from './createMemfs';
import { mockBase64, mockMetroHash } from './mock';

describe('Null component test for @reshoot/loader', () => {
  beforeAll(() => {
    jest.doMock('metrohash', mockMetroHash('tEsThAsH'));
    jest.doMock('../../src/util/base64', mockBase64('data/of=someimage'));
  });

  afterAll((done) => {
    jest.resetModules();
    setTimeout(() => ((done as unknown) as () => void)(), 1000).unref();
  });

  test('should handle null color and background successfully', async () => {
    const memfs = createMemfs();
    const actual = await compile(
      memfs,
      '../../../../__fixtures__/test-image.jpg',
      { outputPath: '/images/', cache: false, color: null, placeholder: null }
    );
    const images = await memfs.promises.readdir(
      path.resolve(__dirname, './images')
    );
    expect(images).toEqual(
      expect.arrayContaining([
        'tEsThAsH-480.jpg',
        'tEsThAsH-640.jpg',
        'tEsThAsH-840.jpg',
        'tEsThAsH-1080.jpg',
        'tEsThAsH-480.webp',
        'tEsThAsH-640.webp',
        'tEsThAsH-840.webp',
        'tEsThAsH-1080.webp',
      ])
    );
    expect(images).toHaveLength(8);
    expect(actual).toMatchSnapshot();
    memfs.reset();
  }, 30000);

  test('should handle null placeholder and aspectRatio successfully', async () => {
    const memfs = createMemfs();
    const actual = await compile(
      memfs,
      '../../../../__fixtures__/test-image.jpg',
      {
        outputPath: '/images/',
        cache: false,
        placeholder: null,
        aspectRatio: null,
      }
    );
    const images = await memfs.promises.readdir(
      path.resolve(__dirname, './images')
    );
    expect(images).toEqual(
      expect.arrayContaining([
        'tEsThAsH-480.jpg',
        'tEsThAsH-640.jpg',
        'tEsThAsH-840.jpg',
        'tEsThAsH-1080.jpg',
        'tEsThAsH-480.webp',
        'tEsThAsH-640.webp',
        'tEsThAsH-840.webp',
        'tEsThAsH-1080.webp',
      ])
    );
    expect(images).toHaveLength(8);
    expect(actual).toMatchSnapshot();
    memfs.reset();
  }, 30000);

  test('should handle empty srcSet and sources successfully', async () => {
    const memfs = createMemfs();
    const actual = await compile(
      memfs,
      '../../../../__fixtures__/test-image.jpg',
      {
        outputPath: '/images/',
        cache: false,
        alternativeWidths: [],
        alternativeFormats: [],
      }
    );
    const images = await memfs.promises.readdir(
      path.resolve(__dirname, './images')
    );
    expect(images).toEqual(['tEsThAsH-840.jpg']);
    expect(actual).toMatchSnapshot();
    memfs.reset();
  }, 30000);
});
