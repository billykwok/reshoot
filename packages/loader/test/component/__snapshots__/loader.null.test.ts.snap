// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Null component test for @reshoot/loader should handle empty srcSet and sources successfully 1`] = `"export default {\\"src\\":__webpack_public_path__+\\"tEsThAsH-840.jpg\\",\\"width\\":4774,\\"height\\":3024,\\"aspectRatio\\":0.6334,\\"placeholder\\":\\"data:image/jpeg;base64,data/of=someimage\\",\\"color\\":\\"#283838\\"};"`;

exports[`Null component test for @reshoot/loader should handle null color and background successfully 1`] = `"export default {\\"src\\":__webpack_public_path__+\\"tEsThAsH-840.jpg\\",\\"width\\":4774,\\"height\\":3024,\\"srcSet\\":__webpack_public_path__+\\"tEsThAsH-480.jpg 480w,\\"+__webpack_public_path__+\\"tEsThAsH-640.jpg 640w,\\"+__webpack_public_path__+\\"tEsThAsH-1080.jpg 1080w\\",\\"aspectRatio\\":0.6334,\\"placeholder\\":\\"data:image/jpeg;base64,data/of=someimage\\",\\"sources\\":[{\\"type\\":\\"image/webp\\",\\"srcSet\\":__webpack_public_path__+\\"tEsThAsH-480.webp 480w,\\"+__webpack_public_path__+\\"tEsThAsH-640.webp 640w,\\"+__webpack_public_path__+\\"tEsThAsH-1080.webp 1080w,\\"+__webpack_public_path__+\\"tEsThAsH-840.webp 840w\\"}]};"`;

exports[`Null component test for @reshoot/loader should handle null placeholder and aspectRatio successfully 1`] = `"export default {\\"src\\":__webpack_public_path__+\\"tEsThAsH-840.jpg\\",\\"width\\":4774,\\"height\\":3024,\\"srcSet\\":__webpack_public_path__+\\"tEsThAsH-480.jpg 480w,\\"+__webpack_public_path__+\\"tEsThAsH-640.jpg 640w,\\"+__webpack_public_path__+\\"tEsThAsH-1080.jpg 1080w\\",\\"aspectRatio\\":0.6334,\\"placeholder\\":\\"data:image/jpeg;base64,data/of=someimage\\",\\"color\\":\\"#283838\\",\\"sources\\":[{\\"type\\":\\"image/webp\\",\\"srcSet\\":__webpack_public_path__+\\"tEsThAsH-480.webp 480w,\\"+__webpack_public_path__+\\"tEsThAsH-640.webp 640w,\\"+__webpack_public_path__+\\"tEsThAsH-1080.webp 1080w,\\"+__webpack_public_path__+\\"tEsThAsH-840.webp 840w\\"}]};"`;
